core/src/test/java/org/apache/iceberg/TestSplitPlanning.java

```diff
@@ -249,6 +249,23 @@ public void testBasicSplitPlanningDeleteFiles() {
     Assert.assertEquals(8, Iterables.size(posDeletesTable.newBatchScan().planTasks()));
   }
 
+  @Test
+  public void testBasicSplitPlanningDeleteFilesWithSplitOffsets() {
+    table.updateProperties().set(TableProperties.FORMAT_VERSION, "2").commit();
+    List<DeleteFile> files128Mb = newDeleteFiles(4, 128 * 1024 * 1024, 8);
+    appendDeleteFiles(files128Mb);
+
+    PositionDeletesTable posDeletesTable = new PositionDeletesTable(table);
+    // we expect 8 bins since split size is 64MB
+    Assert.assertEquals(
```

Comment:
Do you think it will be helpful to add a few more assert, like maybe group the tasks by file, and for each group (2 per group), assert that the second' offset is equal to the first's length.  

Just a suggestion as I dont know enough to know, whether current assert is enough to test fully this code path.

Reply:
> thinking if there is a way to test this i.e finding if these tasks were merged into new task

Trying to follow, are we going to add another test for this case?  (For if the tasks can be merged further, with larger split size?)  I think that is good with me, if I didn't mis-understand you.

Reply:
Extremely sorry missed this comment, let me take a deeper look !

Reply:
@szehon-ho can you please take another pass. Apologies it took me so long to get back.